### DISCUSSION TITLE
convert received webhook customfield to a dictionary

### DISCUSSION THREAD

**Discussion by jnc27** [netbox-community/netbox#13249] [Q&A]
Hi,
Is there a way to easily convert the received custom_fields from webhook to a dictionary?
Webhook receiver config:
```
      {
        "source": "payload",
        "name": "data.custom_fields"
      }

```
I receive it as string: i.e. "map[AS_Number:<nil> Application_Software:<nil> BGP_ASBR:<nil> BGP_RR:<nil>..........
Otherwise I have to retrieve all needed custom_fields one by one:
```
      {
        "source": "payload",
        "name": "data.custom_fields.data1"
      },
      {
        "source": "payload",
        "name": "data.custom_fields.data2l"
      },
      ....
```

Thank you.

**Comment by candlerb** [comment]
Sorry, I think we're missing some essential context for what you're trying to do.

> Is there a way to easily convert the received custom_fields from webhook to a dictionary?
> Webhook receiver config:

Webhook receivers are standalone applications entirely external to Netbox.  Therefore, what application are we talking about?  Is this some code you have written yourself, or is it some existing third-party application which has a config file?

> I receive it as string: i.e. "map[AS_Number: Application_Software: BGP_ASBR: BGP_RR:..........

That to me looks like something generated by the Go programming language. (Aside: Netbox itself is written in Python, although you can write webhooks in whatever language you like)

Therefore, without any further information, I'm guessing you're asking a question about how to decode JSON in a Go program. If so, that is probably best asked on a Go discussion group (e.g. golang-nuts on Google). But maybe I misunderstand you.

If the problem is understanding the format of the JSON payload to a webhook when it has custom fields, then you could try receiving the whole message as a plain string and printing it.  Or at worst, you could use tcpdump to show the raw payload (if you're using HTTP and not HTTPS)
